xonsh/prompt_toolkit_shell.py

```diff
@@ -46,6 +46,11 @@ def __init__(self, **kwargs):
         super().__init__(**kwargs)
         self.history = setup_history()
         self.pt_completer = PromptToolkitCompleter(self.completer, self.ctx)
+        self.mouse_support = builtins.__xonsh_env__.get('MOUSE_SUPPORT')
+        if builtins.__xonsh_env__.get('AUTO_SUGGEST') is True:
+            self.auto_suggest = AutoSuggestFromHistory()
+        else:
+            self.auto_suggest = None
```

Comment:
Also, I think that these lines need to be moved back down to the command loop so that they can be set in the xonshrc file and dynamically turned on/off.  The shell init happens too early.